src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -153,7 +153,15 @@ public class KestrelServerOptions
     /// <summary>
     /// The default server certificate for https endpoints. This is applied lazily after HttpsDefaults and user options.
     /// </summary>
-    internal X509Certificate2? DefaultCertificate { get; set; }
+    /// <remarks>
+    /// Getter exposed for testing.
+    /// </remarks>
+    internal X509Certificate2? DefaultCertificate { get; private set; }
+
+    /// <summary>
+    /// Allow tests to explicitly set the default certificate.
+    /// </summary>
+    internal X509Certificate2? TestOverrideDefaultCertificate { get; set; }
```

Comment:
I split this out because I think it's nice to have invariants on the value of `DefaultCertificate` - in particular, that it's only set if `IsDevCertLoaded` is true.  I could live with letting tests set the real one if people feel strongly.

Reply:
So DefaultCertificate should be renamed to DevCertificate?

Reply:
Since you added these other properties to distinguish where the cert is coming from, it does make sense to rename DefaultCertificate to DeveloperCertificate.